syntax = "proto2";

package bigquery.utils.zetasql_helper.local_service;

import "zetasql_helper/token/parse_token.proto";

option java_package = "com.google.bigquery.utils.zetasqlhelper";
option java_outer_classname = "LocalService";

service ZetaSqlHelperLocalService {

  rpc Hello(HelloRequest) returns (HelloResponse) {
  }

  rpc Tokenize(TokenizeRequest) returns (TokenizeResponse) {
  }

}

message HelloRequest {
  optional string name = 1;
}

message HelloResponse {
  optional string greeting = 1;
}

message TokenizeRequest {
  optional string query = 1;
}

message TokenizeResponse {
  repeated ParseTokenProto parse_token = 1;
}